pages/committee/[period-id]/[committee]/index.tsx

```diff
@@ -0,0 +1,214 @@
+
+      try {
+        const response = await fetch(
+          `/api/committees/times/${period?._id}/${committee}`
+        );
+        const data = await response.json();
+        if (response.ok) {
+          setCommitteeInterviewTimes(data.committees[0]);
+        } else {
+          throw new Error(data.error || "Unknown error");
+        }
+      } catch (error) {
+        console.error("Error checking period:", error);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchCommitteeInterviewTimes();
+  }, [tabClicked, period]);
+
+  useEffect(() => {
+    if (!session || !periodId || !committee) return;
+
+    const checkAccess = () => {
+      if (!period) {
+        return;
+      }
+      const userCommittees = session?.user?.committees?.map((committee) =>
+        committee.toLowerCase()
+      );
+      const periodCommittees = period.committees.map((committee) =>
+        committee.toLowerCase()
+      );
+
+      period.optionalCommittees.forEach((committee) => {
+        periodCommittees.push(committee.toLowerCase());
+      });
+
+      const commonCommittees = userCommittees!.filter((committee) =>
+        periodCommittees.includes(committee)
+      );
+      if (commonCommittees.includes(committee)) {
+        setHasAccess(true);
+      } else {
+        setLoading(false);
+      }
+    };
+
+    checkAccess();
+  }, [period]);
+
+  if (loading) {
+    return <LoadingPage />;
+  }
+
+  if (!session || !hasAccess) {
+    return <Custom404 />;
+  }
+
+  const interviewPeriodEnd = period?.interviewPeriod.end
+    ? new Date(period.interviewPeriod.end)
+    : null;
+
+  //Satt frist til 14 dager etter intervju perioden, så får man ikke tilgang
+  if (
+    interviewPeriodEnd &&
+    interviewPeriodEnd.getTime() + 14 * 24 * 60 * 60 * 1000 <
+      new Date().getTime()
+  ) {
+    return (
+      <div className="flex flex-col h-screen text-center items-center justify-center px-20">
+        <h1 className="text-3xl">Opptaket er ferdig!</h1>
+        <br />
+        <p className="text-lg">
+          Du kan ikke lenger se søkere eller planlegge intervjuer.
+        </p>
+        <p className="text-lg">
+          {" "}
+          Har det skjedd noe feil eller trenger du tilgang til informasjonen? Ta
+          kontakt med{" "}
+          <a
+            className="font-semibold underline transition-all hover:text-online-orange"
+            href="mailto:appkom@online.ntnu.no"
+          >
+            Appkom
+          </a>{" "}
+        </p>
+      </div>
+    );
+  }
+
+  return (
+    <div className="flex flex-col items-center">
+      <span className="mt-5 mb-6 text-3xl font-bold text-center">
+        <h1>
+          <button onClick={() => router.push("/committee")}>
+            <span>{period?.name}</span>
+          </button>
+          <span>{` --> `}</span>
+          <button onClick={() => router.push(`/committee/${periodId}`)}>
+            <span>{changeDisplayName(committee)}</span>
+          </button>
```

Comment:
synes dette burde gjøres på  en annen måte, men kan godt gjøre dette på en annen branch hvor jeg også fikser styling?

Reply:
Ja enig at stylingen er ganske stygt

Reply:
> synes dette burde vises på en annen måte, men kan godt gjøre dette på en annen branch hvor jeg også fikser styling?

Enig også at det er bedre å fikse styling i en annen branch, med tanke på at denne er en blocker for andre issues